google/cloud/bigtable/examples/table_admin_snippets.cc

```diff
@@ -358,6 +358,91 @@ void GetFamilyMetadata(google::cloud::bigtable::TableAdmin admin, int argc,
   (std::move(admin), table_id, family_name);
 }
 
+void DeleteColumnFamily(google::cloud::bigtable::TableAdmin admin, int argc,
+                        char* argv[]) {
+  if (argc != 3) {
+    throw Usage{
+        "delete-column-family: <project-id> <instance-id> <table-id>"
+        " <family-name>"};
+  }
+  std::string const table_id = ConsumeArg(argc, argv);
+  std::string const family_name = ConsumeArg(argc, argv);
+
+  // [START bigtable_delete_family]
+  namespace cbt = google::cloud::bigtable;
+  [](cbt::TableAdmin admin, std::string table_id, std::string family_name) {
+    auto schema = admin.ModifyColumnFamilies(
+        table_id, {cbt::ColumnFamilyModification::Drop(family_name)});
+
+    if (!schema) {
+      throw std::runtime_error(schema.status().message());
+    }
+    std::string formatted;
+    google::protobuf::TextFormat::PrintToString(*schema, &formatted);
+    std::cout << "Schema modified to: " << formatted << "\n";
+  }
+  // [END bigtable_delete_family]
+  (std::move(admin), table_id, family_name);
+}
+
+void CheckFamilyExists(google::cloud::bigtable::TableAdmin admin, int argc,
+                       char* argv[]) {
+  if (argc != 3) {
+    throw Usage{
+        "check-family-exists: <project-id> <instance-id> <table-id>"
+        " <family-name>"};
+  }
+  std::string const table_id = ConsumeArg(argc, argv);
+  std::string const family_name = ConsumeArg(argc, argv);
+
+  // [START bigtable_check_family_exists]
+  namespace cbt = google::cloud::bigtable;
+  [](cbt::TableAdmin admin, std::string table_id, std::string family_name) {
+    auto schema =
+        admin.GetTable(table_id, google::bigtable::admin::v2::Table::FULL);
+
+    if (!schema) {
+      throw std::runtime_error(schema.status().message());
+    }
+    auto pos = schema->column_families().find(family_name);
+    if (pos == schema->column_families().end()) {
+      std::cout << "The column family <" << family_name << "> does not exist\n";
```

Comment:
Shouldn't this fail somehow?

Reply:
The sample is "bigtable_check_family_exists" I guessed that "not exists" is a perfectly normal result for that example. @billyjacobson may be able to clarify.

Reply:
I ask because running the example (from run_examples_utils) really then only "checks" that the table exists.  Whether the column family exists doesn't matter.  Having "check_family_exists" exit with a non-success code in the "not exists" case (with this std::cout possibly being std::cerr) sounds like a feature.